### PR TITLE
added cctk

### PR DESCRIPTION
**Pull request by corinwagen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

I think I've done everything correctly here? Apologies if not. 

( @conda-forge-admin, please ping team )

### CODE REVIEW

**Comment by corinwagen** [comment]
Sorry, added ``LICENSE`` to cctk's ``MANIFEST.in``

File: recipes/cctk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{% set name = "cctk" %}
+{% set version = "0.2.18" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
```
Comment (ocefpaf) [inline recipes/cctk/meta.yaml:9]:
```suggestion
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

File: recipes/cctk/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  sha256: 83881c5ce07edcb5b45b9e4d7771992974c1cf622d8ef272d4f35e9f28889037
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation"
```
Comment (ocefpaf) [inline recipes/cctk/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/cctk/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  host:
+    - basis_set_exchange
+    - importlib_resources
+    - networkx
+    - numpy
+    - pip
+    - pyahocorasick
+    - python >=3.7
+    - pyyaml
+    - scipy
```
Comment (ocefpaf) [inline recipes/cctk/meta.yaml:27]: You don't need all the runtime dependencies at build time, unless the package imports itself at build time, then you should report that upstream. BTW, the metadata says `python_requires='>=3.6'`, not 3.7.

```suggestion
    - python >=3.6
    - pip
```

File: recipes/cctk/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  run:
+    - basis_set_exchange
+    - importlib_resources
+    - networkx
+    - numpy
+    - pyahocorasick
+    - python >=3.7
```
Comment (ocefpaf) [inline recipes/cctk/meta.yaml:34]:
```suggestion
    - python >=3.6
```

File: recipes/cctk/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    - pyyaml
+    - scipy
+
+test:
+  imports:
+    - cctk
+    - cctk.data
+    - cctk.groups
```
Comment (ocefpaf) [inline recipes/cctk/meta.yaml:42]:
```suggestion
    - cctk.groups
  commands:
    - pip check
  requires:
    - pip
```

File: recipes/cctk/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+about:
+  home: "https://github.com/ekwan/cctk"
```
Comment (ocefpaf) [inline recipes/cctk/meta.yaml:45]:
```suggestion
  home: https://github.com/ekwan/cctk
```

File: recipes/cctk/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  license: Apache-2.0
+  license_family: Apache
+  license_file: LICENSE
+  summary: "computational chemistry toolkit"
```
Comment (ocefpaf) [inline recipes/cctk/meta.yaml:49]:
```suggestion
  summary: computational chemistry toolkit
```

**Review by ocefpaf — COMMENTED** [review]
Please use [grayskull](https://github.com/conda-forge/staged-recipes#grayskull---recipe-generator-for-python-packages-on-pypi) when creating Python recipes. It makes the review much easier and creates a recipe that is almost "merge-ready."

**Comment by corinwagen** [comment]
Ok, thanks for the help! I used ``conda skeleton`` but will use grayskull in future, thanks.

**Comment by Alan-Collins** [comment]
@corinwagen @ocefpaf I am the developer of the cctk package on Bioconda. https://anaconda.org/bioconda/cctk

The conda-forge bot caught that there was already a cctk package during the original PR process https://github.com/conda-forge/staged-recipes/pull/24210#issuecomment-1758066222

Can we rename the cctk package on conda-forge?

**Comment by ocefpaf** [comment]
> Can we rename the cctk package on conda-forge?

We definitely should. I probably did not check and thought it was the same :-/

@corinwagen, can you do that?

**Comment by ocefpaf** [comment]
> I'm happy to help here; how do I do that?

The cleanest way would be to:

1. mark all packages as broken: https://github.com/conda-forge/admin-requests?tab=readme-ov-file#mark-packages-as-broken-on-conda-forge
2. archive the feedstock: https://github.com/conda-forge/admin-requests?tab=readme-ov-file#archive-or-unarchive-a-feedstock
3. re-submit to staged-recipes with the new name

**Comment by ocefpaf** [comment]
It will, those will need to be updated. We could argue that the bioconda one should change b/c this one is on PyPI and all... However, the bioconda one was here first.

**Comment by Alan-Collins** [comment]
I have no plans to put my package on pypi so there is no conflict there.

Do you know how many conda-forge packages depend on the cctk on conda-forge? My bioconda packages is not a dependency of any other bioconda package. If renaming the conda-forge package would break other packages then I would be willing to rename the bioconda package.

My primary issue is communicating a name change to users, but that is an issue we both face. I'm happy to take the path that is least disruptive here.

**Comment by ocefpaf** [comment]
> Do you know how many conda-forge packages depend on the cctk on conda-forge?

Looks like none: https://github.com/search?q=org%3Aconda-forge%20path%3Ameta.yaml%20cctk&type=code

**Comment by Alan-Collins** [comment]
In that case would it be acceptable if we rename the conda-forge package, but leave the pypi package as it is? It looks like @corinwagen has some other pypi packages built off the pypi cctk.

**Comment by ocefpaf** [comment]
> In that case would it be acceptable if we rename the conda-forge package, but leave the pypi package as it is? It looks like @corinwagen has some other pypi packages built off the pypi cctk.

We can't tell you what to do with PyPI. It is your package 😄 .

Also, to have some connection with it, I would recommend you calling it `python-cctk`, so people know that it is on PyPI with the `cctk` name.

**Comment by Alan-Collins** [comment]
Sorry for the confusion. I'm not the one with the pypi package. I'm the dev of the Bioconda package. I was just trying to confirm how we are moving forward.

**Comment by corinwagen** [comment]
Yep, I'm the one with the `conda-forge` package and the `pypi` package. 

I'm stalling a bit because we have a (non-public) [`pixi`](https://prefix.dev/) build that depends on `cctk` being in `conda-forge` for a bunch of AWS EC2 workers that constantly get spun up, so we'll have to redeploy our production infrastructure if we switch the name to `cctk-python`, but our lead devops engineer is traveling for the holidays right now so I'm a bit hesitant to start breaking things. ([I'm the CEO of a computational chemistry startup](https://rowansci.com/about); we use `cctk` for a variety of things internally.)

@Alan-Collins is it okay if this waits a little bit for you? what's the urgency level on your end?

**Comment by Alan-Collins** [comment]
That makes sense. It's fine if this waits until after the holidays. Thank you!

**Comment by Alan-Collins** [comment]
Hey @corinwagen, just wanted to check in about the progress of renaming your conda-forge package?

**Comment by Alan-Collins** [comment]
@ocefpaf can you confirm these details  please?

> ah, yes, let me do this. so first I submit them as broken following [this example](https://github.com/conda-forge/admin-requests/blob/main/examples/example-broken.yml), then I submit an archive-feedstock [like this](https://github.com/conda-forge/admin-requests/blob/main/examples/example-archive.yml), then I re-submit to staged recipes with a new name (not `cctk`)?
> 
> do these PRs go to `staged-recipes` or `admin-requests`?

**Comment by corinwagen** [comment]
Just initialized the process: https://github.com/conda-forge/admin-requests/pull/1421

**Comment by corinwagen** [comment]
Yeah, for sure! Sorry that it took me a sec

**Pull request merged**